plugin/omero_iviewer/urls.py

```diff
@@ -1,13 +1,34 @@
+#
+# Copyright (c) 2017 University of Dundee.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
 from django.conf.urls import patterns
 from django.conf.urls import url
 
 import views
 
-urlpatterns = patterns('django.views.generic.simple',
-                       url(r'^(?P<iid>[0-9]+)?/?$', views.index,
-                           name='omero_iviewer_index'),
-                       url(r'^persist_rois/?$', views.persist_rois,
-                           name='omero_iviewer_persist_rois'),
-                       url(r'^request_rois/(?P<iid>[0-9]+)?/?$',
-                           views.request_rois,
-                           name='omero_iviewer_request_rois'))
+urlpatterns = patterns(
+
+    'django.views.generic.simple',
+
+    # index 'home page' of the iviewer app
+    url(r'^(?P<iid>[0-9]+)?/?$', views.index, name='omero_iviewer_index'),
+    url(r'^persist_rois/?$', views.persist_rois,
+        name='omero_iviewer_persist_rois'),
+    url(r'^request_rois/(?P<iid>[0-9]+)?/?$', views.request_rois,
+        name='omero_iviewer_request_rois'),
+    url(r'^image_data/(?P<image_id>[0-9]+)?/?$', views.image_data,
```

Comment:
This should be ``` url(r'^image_data/(?P<image_id>[0-9]+)/$' ``` since the ```image_id``` is not optional.
The same can be applied to a few other urls that have used the same pattern.